hoomd/md/NeighborList.cc

```diff
@@ -366,6 +366,19 @@ void NeighborList::compute(uint64_t timestep)
     if (m_prof)
         m_prof->push("Neighbor");
 
+    // when the number of particles in the system changes, rebuild the exclusion list
+    if (m_need_reallocate_exlist)
+        {
+        clearExclusions();
+        std::set<std::string> exclusions_copy(m_exclusions);
+        for (const std::string& exclusion : exclusions_copy)
```

Comment:
Since we are not editing exclusions is there a reason we aren't iterating over `m_exclusions` directly?

Reply:
`setSingleExclusion` inserts `exclusion` into `m_exclusions`. I do not know whether inserting an existing key into a `set` invalidates iterators, so I iterate over a copy for safety.

Reply:
Insertion invalidates iterators for sequential containers but not for associative containers according to [the standard, 23.2.4/9](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4296.pdf) (I picked C++14, but I believe it holds even in C++03). That said, this code looks fine to me; it's not performance critical and I see no reason to rely on that property of the containers.